### DISCUSSION TITLE
border router cannot route ping response when nat64 prefix is registered in the network

### DISCUSSION THREAD

**Discussion by zimmele2** [openthread/openthread#7861] [Q&A] [answered]
Hello,

I have set up a thread network with a border router and a single thread device with Zephyr (v2.7).
In order for my thread device to push data to the server, it needs to resolve a DNS query for an IPv4 address first.
I configured tayga on my border router and that seems to work. I can ping any ipv4 address with the nat64 prefix i defined and I can also make a DNS query, which seems to work when you look at the tcp dump but on the thread device the response is empty:

tcp dump on nat64 interface of dns query:
```
15:21:20.137096 IP6 (hlim 63, next-header UDP (17) payload length: 33) fda7:af2b:774a:1:f163:7c2b:4d15:1f03.49153 > fdde:efde:efde:deef:1234:0:808:808.domain: [udp sum ok] 22162+ AAAA? zhaw.ch. (25)
15:21:20.150715 IP6 (hlim 112, next-header UDP (17) payload length: 84) fdde:efde:efde:deef:1234:0:808:808.domain > fda7:af2b:774a:1:f163:7c2b:4d15:1f03.49153: [udp sum ok] 22162 0/1/0 (76)
15:21:20.170139 IP6 (hlim 63, next-header UDP (17) payload length: 33) fda7:af2b:774a:1:f163:7c2b:4d15:1f03.49153 > fdde:efde:efde:deef:1234:0:808:808.domain: [udp sum ok] 42425+ A? zhaw.ch. (25)
15:21:20.187859 IP6 (hlim 112, next-header UDP (17) payload length: 49) fdde:efde:efde:deef:1234:0:808:808.domain > fda7:af2b:774a:1:f163:7c2b:4d15:1f03.49153: [udp sum ok] 42425 1/0/0 zhaw.ch. A 160.85.104.112 (41)
```

thread device dns query:
```
uart:~$ ot dns resolve zhaw.ch
DNS response for zhaw.ch. - 
Done
```

The problem is that when i do a dns query for a IPv4 `GetNat64Prefix` returns `kErrorNotFound` because none of the prefixes is a nat64 prefix. So i registered my nat64 prefix with the border router including the nat64 flag: 
`sudo ot-ctl prefix remove fdde:efde:efde:deef:1234::/96 n`
`sudo ot-ctl netdata register`

With the nat64 prefix pinging doesnt work anymore. I can see the response in the tcp dump from the nat64 interface, but not on the thread device.

tcpdump on nat64 interface:
```
15:35:01.999511 IP6 (hlim 63, next-header ICMPv6 (58) payload length: 16) fda7:af2b:774a:1:f163:7c2b:4d15:1f03 > fdde:efde:efde:deef:1234:0:a00:cb: [icmp6 sum ok] ICMP6, echo request, id 18, seq 18
15:35:01.999948 IP6 (hlim 63, next-header ICMPv6 (58) payload length: 16) fdde:efde:efde:deef:1234:0:a00:cb > fda7:af2b:774a:1:f163:7c2b:4d15:1f03: [icmp6 sum ok] ICMP6, echo reply, id 18, seq 18

```

thread device ping:
```
uart:~$ ot ping fdde:efde:efde:deef:1234::8.8.8.8
1 packets transmitted, 0 packets received. Packet loss = 100.0%.
Done
```

Why can I not receive responses on my thread device when the nat64 prefix is registered within the network?
How can I register the nat64 prefix in the network, so that my thread device can synthesize the ipv6 address out of my prefix and the dns response (ipv4)?

I attached some additional info in the netdata.txt file, dont know if its of any help.

Thank you

[netdata.txt](https://github.com/openthread/openthread/files/9029781/netdata.txt)

**Comment by jwhui** [comment, marked as answer]
To add a NAT64 prefix with the CLI, use the [`route add`](https://github.com/openthread/openthread/tree/main/src/cli#route-add-prefix-snprf) command.

With your example:
```
> route add fdde:efde:efde:deef:1234::/96 n
> netdata register
```

**Reply by zimmele2** [reply]
Thank you, that worked
